crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -447,7 +448,11 @@ fn stub_file_empty_lines(
         }
         SuiteKind::Class | SuiteKind::Other | SuiteKind::Function => {
             if empty_line_condition
-                && lines_after_ignoring_end_of_line_trivia(preceding.end(), source) > 1
+                && ((is_blank_line_after_nested_stub_class_enabled(f.context())
```

Comment:
Should this also apply if the class is nested in a if/while or any other control flow block?

Reply:
Yes